src/repositories/request-repository.ts

```diff
@@ -37,327 +81,256 @@ const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void =
 }
 
 @singleton()
-@EntityRepository(Request)
-export class RequestRepository extends Repository<Request> {
-  private readonly isolationLevel
-
-  constructor(
-    @inject('config') private config?: Config,
-    @inject('dbConnection') private connection?: Connection
-  ) {
-    super()
-    this.isolationLevel =
-      this.connection.options.type === 'sqlite' ? 'SERIALIZABLE' : 'REPEATABLE READ'
-  }
+export class RequestRepository {
+  constructor(@inject('dbConnection') private connection?: Knex) {}
 
   /**
    * Create/updates client request
    * @param request - Request
+   * @param options
+   * @returns A promise that resolves to the created request
+   */
+  public async createOrUpdate(request: Request, options: Options = {}): Promise<Request> {
+    const { connection = this.connection } = options
+    const [{ id }] = await connection
+      .table(TABLE_NAME)
+      .insert(request, ['id'])
+      .onConflict('cid')
+      .merge()
+
+    const created = await connection.table(TABLE_NAME).first().where({ id })
+
+    logEvent.db({
```

Comment:
I can't find where this happened before.  Is this new?

Reply:
We used to have a file: `src/subscribers/request-subscriber.ts` that supposedly logged on every request update and creation. I didn't actually see it working even tho we did use it in the typeorm configs. We have an in-code log on every update so I added a log event for creation just in case we actually wanted those logs but they were broken before and no one noticed. I can remove this if its not neccessary